public/app/features/explore/Logs/Logs.tsx

```diff
@@ -669,31 +683,32 @@ const UnthemedLogs: React.FunctionComponent<Props> = (props: Props) => {
       updateItem(logsParent.id, { expanded: true });
     }
 
-    register?.({
-      icon: 'gf-logs',
-      title: 'Pinned log',
-      panelId: 'Logs',
-      level: 'child',
-      ref: null,
-      color: LogLevelColor[row.logLevel],
-      childOnTop: true,
-      onClick: () => onOpenContext(row, () => {}),
-      onRemove: (id: string) => {
-        unregister?.(id);
-        if (getPinnedLogsCount() < PINNED_LOGS_LIMIT) {
-          setPinLineButtonTooltipTitle('Pin to content outline');
-        }
-      },
-    });
+    const alreadyPinned = pinnedLogs?.find((pin) => pin.id === row.rowId);
+    if (alreadyPinned && row.rowId && allowUnPin) {
+      unregister?.(row.rowId);
+    } else if (getPinnedLogsCount() !== PINNED_LOGS_LIMIT) {
```

Comment:
I think this should only get called if the line is not pinned already.
```suggestion
    } else if (getPinnedLogsCount() !== PINNED_LOGS_LIMIT && !alreadyPinned) {
```


https://github.com/grafana/grafana/assets/8092184/c0aeaece-41df-4ebc-a325-6edb6a163d0b